src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Completion/Delegation/DelegatedCompletionListProviderTest.cs

```diff
@@ -122,130 +131,50 @@ public async Task HtmlDelegation_UnsupportedTriggerCharacter_TranslatesToInvoked
-            // Arrange
-            var completionContext = new VSInternalCompletionContext() { TriggerKind = CompletionTriggerKind.Invoked };
-            var codeDocument = CreateCodeDocument("@functions ");
-            var documentContext = TestDocumentContext.From("C:/path/to/file.razor", codeDocument, hostDocumentVersion: 1337);
-
-            // Act
-            var completionList = await Provider.GetCompletionListAsync(absoluteIndex: 11, completionContext, documentContext, ClientCapabilities, CancellationToken.None);
+            // Arrange & Act
+            var completionList = await GetCompletionListAsync("@$$", CompletionTriggerKind.Invoked);
 
             // Assert
-            Assert.Null(completionList);
-            var delegatedParameters = Provider.DelegatedParams;
-            Assert.Null(delegatedParameters);
+            Assert.Contains(completionList.Items, item => item.Label == "System");
         }
 
         [Fact]
-        public async Task CSharpDelegation_TriggerCharacterAt_TranslatesToInvoked()
+        public async Task CSharp_At_TranslatesToInvoked_Triggered()
         {
-            // Arrange
-            var completionContext = new VSInternalCompletionContext()
-            {
-                InvokeKind = VSInternalCompletionInvokeKind.Typing,
-                TriggerKind = CompletionTriggerKind.TriggerCharacter,
-                TriggerCharacter = "@",
-            };
-            var codeDocument = CreateCodeDocument("@");
-            var documentContext = TestDocumentContext.From("C:/path/to/file.cshtml", codeDocument, hostDocumentVersion: 1337);
-
-            // Act
-            await Provider.GetCompletionListAsync(absoluteIndex: 1, completionContext, documentContext, ClientCapabilities, CancellationToken.None);
+            // Arrange & Act
+            var completionList = await GetCompletionListAsync("@$$", CompletionTriggerKind.TriggerCharacter);
 
             // Assert
-            var delegatedParameters = Provider.DelegatedParams;
-            Assert.NotNull(delegatedParameters);
-            Assert.Equal(RazorLanguageKind.CSharp, delegatedParameters.ProjectedKind);
-
-            // Just validating that we're generating code in a way that's different from the top-level document. Don't need to be specific.
-            Assert.True(delegatedParameters.ProjectedPosition.Line > 2);
-            Assert.Equal(CompletionTriggerKind.Invoked, delegatedParameters.Context.TriggerKind);
-            Assert.Equal(VSInternalCompletionInvokeKind.Explicit, delegatedParameters.Context.InvokeKind);
-            Assert.Equal(1337, delegatedParameters.HostDocument.Version);
-            Assert.Null(delegatedParameters.ProvisionalTextEdit);
+            Assert.Contains(completionList.Items, item => item.Label == "System");
         }
 
-        [Fact]
-        public async Task CSharpDelegation_TriggerCharacter()
+        [Fact(Skip = "For some reason trigger based completion items aren't returning any items")]
```

Comment:
@allisonchou / @dibarbet I couldn't figure out why the C# completion handler wasn't respecting trigger character based completions. Any ideas pop to the top of your head?

If I swapped this to "CompletionTriggerKind.Invoked" then all works well (well minus some expected completions like `DateTime`)

Reply:
Oh interesting, seems like this might have been broken for a while (if it ever worked at all). Just tried on 17.2.5 and completion didn't show up.

I can try to debug locally later today to see what I can find, off the top of my head I don't have any ideas